opentelemetry-sdk/src/opentelemetry/sdk/_metrics/metric_reader_storage.py

```diff
@@ -44,27 +50,58 @@ def _get_or_init_view_instrument_match(
         # Optimistically get the relevant views for the given instrument. Once set for a given
         # instrument, the mapping will never change
 
-        if instrument in self._view_instrument_match:
-            return self._view_instrument_match[instrument]
+        if instrument in self._instrument_view_instrument_matches:
+            return self._instrument_view_instrument_matches[instrument]
 
         with self._lock:
             # double check if it was set before we held the lock
-            if instrument in self._view_instrument_match:
-                return self._view_instrument_match[instrument]
+            if instrument in self._instrument_view_instrument_matches:
+                return self._instrument_view_instrument_matches[instrument]
 
             # not present, hold the lock and add a new mapping
             view_instrument_matches = []
             for view in self._sdk_config.views:
                 # pylint: disable=protected-access
                 if view._match(instrument):
-                    view_instrument_matches.append(
-                        _ViewInstrumentMatch(
-                            view=view,
-                            instrument=instrument,
-                            sdk_config=self._sdk_config,
-                        )
+
+                    new_view_instrument_match = _ViewInstrumentMatch(
+                        view=view,
+                        instrument=instrument,
+                        sdk_config=self._sdk_config,
                     )
 
+                    if isinstance(instrument, Asynchronous) and isinstance(
+                        view._aggregation, ExplicitBucketHistogramAggregation
+                    ):
+                        _logger.warning(
+                            "View %s and instrument %s will produce "
+                            "semantic errors when matched, the view "
+                            "has not been applied.",
+                            view,
+                            instrument,
+                        )
+                        continue
+
+                    for (
+                        existing_view_instrument_matches
+                    ) in self._instrument_view_instrument_matches.values():
+                        for (
+                            existing_view_instrument_match
+                        ) in existing_view_instrument_matches:
+                            if (
+                                existing_view_instrument_match._name
+                                == new_view_instrument_match._name
```

Comment:
Were we not going to include the other identifying properties for comparison as defined [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#opentelemetry-protocol-data-model)?

Reply:
Added the remaining properties for consideration.